plugins/voidscape/skills/voidscape/scripts/observe.py

```
@@ -0,0 +1,478 @@
+        parsed.port
+    except ValueError as ex:
+        raise _input_error("screenpipe URL has an invalid port") from ex
+    if parsed.path not in {"", "/"} or parsed.query or parsed.fragment:
+        raise _input_error("screenpipe URL must be a loopback origin without a path or query")
+    return value.rstrip("/")
+
+
+def _fetch_health(url: str) -> dict[str, Any] | None:
+    request = Request(f"{url}/health", headers={"Accept": "application/json"})
+    try:
+        with urlopen(request, timeout=2) as response:
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)